server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -155,6 +155,10 @@ public boolean isSearchable() {
         return isIndexed;
     }
 
+    public boolean hasDocValuesSparseIndex() {
```

Comment:
I think for now, we don't have to add a new method to this base class? I see this is only used in tests and then we know the concrete class?